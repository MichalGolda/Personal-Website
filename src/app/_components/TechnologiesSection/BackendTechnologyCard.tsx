import TechnologyCard from "./TechnologyCard";
import {
  AWSIcon,
  AzureIcon,
  DjangoIcon,
  DockerIcon,
  GCPIcon,
  JavascriptIcon,
  NestJsIcon,
  NodeJsIcon,
  PostgreSQLIcon,
  TypescriptIcon,
} from "./TechnologyIcon";

const Icons = () => {
  return (
    <>
      <TypescriptIcon />
      <JavascriptIcon />
      <DjangoIcon />
      <DockerIcon />
      <NodeJsIcon />
      <NestJsIcon />
      <PostgreSQLIcon />
      <AWSIcon />
      <GCPIcon />
      <AzureIcon />
    </>
  );
};

const BackendTechnologyCard = () => {
  return (
    <TechnologyCard
      title="Backend"
      content={() => (
        <>
          Dostępność, skalowalność osiągam dzięki sprawdzonym rozwiązaniom:{" "}
          <b className="text-primary">Python</b>,{" "}
          <b className="text-primary">Django</b>,{" "}
          <b className="text-primary">PHP</b>,{" "}
          <b className="text-primary">Wordpress</b>,{" "}
          <b className="text-primary">Docker</b>,{" "}
          <b className="text-primary">NodeJs</b>,{" "}
          <b className="text-primary">Nest.js</b>,{" "}
          <b className="text-primary">PostgreSQL</b>,{" "}
          <b className="text-primary">AWS</b>,{" "}
          <b className="text-primary">GCP</b>,{" "}
          <b className="text-primary">Azure</b>.
        </>
      )}
      icons={<Icons />}
    />
  );
};

export default BackendTechnologyCard;
