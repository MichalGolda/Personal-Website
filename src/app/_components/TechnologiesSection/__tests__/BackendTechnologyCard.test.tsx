import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import BackendTechnologyCard from "../BackendTechnologyCard";

describe("BackendTechnologyCard component", () => {
  it("renders unchanged", () => {
    const { container } = render(<BackendTechnologyCard />);

    expect(container).toMatchSnapshot();
  });
});
