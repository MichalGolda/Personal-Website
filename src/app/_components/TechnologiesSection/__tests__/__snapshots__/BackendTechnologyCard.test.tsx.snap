// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackendTechnologyCard component renders unchanged 1`] = `
<div>
  <div
    class="w-full flex flex-col gap-y-4"
  >
    <h4
      class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
    >
      Backend
    </h4>
    <p
      class="text-sm md:text-base text-body"
    >
      Dostępność, skalowalność osiągam dzięki sprawdzonym rozwiązaniom:
       
      <b
        class="text-primary"
      >
        Python
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Django
      </b>
      ,
       
      <b
        class="text-primary"
      >
        PHP
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Wordpress
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Docker
      </b>
      ,
       
      <b
        class="text-primary"
      >
        NodeJs
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Nest.js
      </b>
      ,
       
      <b
        class="text-primary"
      >
        PostgreSQL
      </b>
      ,
       
      <b
        class="text-primary"
      >
        AWS
      </b>
      ,
       
      <b
        class="text-primary"
      >
        GCP
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Azure
      </b>
      .
    </p>
    <div
      class="flex flex-row flex-wrap gap-4"
    >
      <img
        alt="typescript"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Ftypescript.webp&w=64&q=75"
        srcset="/_next/image?url=%2Ftypescript.webp&w=32&q=75 1x, /_next/image?url=%2Ftypescript.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="javascript"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fjavascript.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fjavascript.webp&w=32&q=75 1x, /_next/image?url=%2Fjavascript.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="django"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fdjango.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fdjango.webp&w=32&q=75 1x, /_next/image?url=%2Fdjango.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="docker"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fdocker.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fdocker.webp&w=32&q=75 1x, /_next/image?url=%2Fdocker.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="nodejs"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fnodejs.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fnodejs.webp&w=32&q=75 1x, /_next/image?url=%2Fnodejs.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="nestjs"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fnestjs.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fnestjs.webp&w=32&q=75 1x, /_next/image?url=%2Fnestjs.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="postgresql"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fpostgresql.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fpostgresql.webp&w=32&q=75 1x, /_next/image?url=%2Fpostgresql.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="aws"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Faws.webp&w=64&q=75"
        srcset="/_next/image?url=%2Faws.webp&w=32&q=75 1x, /_next/image?url=%2Faws.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="gcp"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fgcp.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fgcp.webp&w=32&q=75 1x, /_next/image?url=%2Fgcp.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="azure"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fazure.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fazure.webp&w=32&q=75 1x, /_next/image?url=%2Fazure.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
    </div>
  </div>
</div>
`;
